src/host.c

```diff
@@ -626,6 +626,44 @@ cache_store (const char *host, struct address_list *al)
     }
 }
 
+bool
+resolve (const char *host, const char *ip)
+{
+  size_t len = strnlen(ip, 45);
+  bool validIpv4 = is_valid_ipv4_address(ip, ip + len);
+  bool validIpv6 = is_valid_ipv6_address(ip, ip + len);
+
+  if (validIpv4 == validIpv6) {
+    logprintf(LOG_ALWAYS, "Error: Not a valid IP address or an ambiguius IPv4/IPv6\n");
+    return false;
```

Comment:
C-community will crucify you for using multiple exit points ;)

Reply:
As far as I see there can be multiple "error" exit points, with only one "positive" one, what do you think?